runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/catalog/CatalogHandler.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc.
+ *
+ * Aklivity licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.aklivity.zilla.runtime.engine.catalog;
+
+public interface CatalogHandler
+{
+    <T> T register(String schemaDefinition, String subject, String dataFormat);
+
+    String getSchemasById(String schemaId);
+
+    String getSchemaBySubject(String subject);
+
+    String getVersionSchema(String subject, String version);
```

Comment:
`schemaId` should be an `int`, right?

The purpose of register is to create an official `id` for the new schema, so:
```
  int register(String name, String schema, String format);
```
Potentially don't need format if it can be discerned from the schema definition.

The get-style methods don't fit with the rest of our internal APIs, suggest we use `resolve` instead.
```
  String resolve(int schemaId);
  int resolve(String name, String version); // default version is "latest"
```

So we likely only need 3 methods here.